server/tests/unit/common/config/test_dataset_config.py

```diff
@@ -119,19 +120,28 @@ def test_multi_dataset(self):
         server.testing = True
         session = server.test_client()
 
+        def _get_v03_url(url):
+            response = session.get(f"{url}/api/v0.3/s3_uri")
+            s3_uri = quote(quote(response.json, safe=""), safe="")
+            return f"/s3_uri/{s3_uri}/api/v0.3"
+
+
         with self.subTest("Test config for dataroot /set1/1/2/ returns the s1 config"):
-            response1 = session.get("/set1/1/2/pbmc3k.cxg/api/v0.2/config")
+            v03_url = _get_v03_url("/set1/1/2/pbmc3k.cxg")
+            response1 = session.get(f"{v03_url}/config")
```

Comment:
It's preferable to specify full literal values in tests, for readability.  Being DRY here and/or computing test-related values isn't removing much code, but it is forcing reader to skip around.

If we're only testing the methods that take an s3_uri, and not testing the s3_uri lookup api call it self, I would make it clearer in these tests by doing the s3_uri lookup in a setup method (or commented section), or even just hardcode the s3_uri (it's deterministic). If we _are_ testing the s3_uri lookup, I wouldn't relegate it to a method; I would keep it inlined in the test. Or test the s3_uri lookup directly in its own test.

Reply:
hard coding is difficult since it's the full path to the fixture. I'll add comments